src/csync/csync_private.h

```diff
@@ -148,17 +147,13 @@ struct OCSYNC_EXPORT csync_s {
 
   OCC::SyncJournalDb *statedb;
 
-  c_strlist_t *excludes = nullptr; /* list of individual patterns collected from all exclude files */
-  struct TraversalExcludes {
-      ~TraversalExcludes() {
-          c_strlist_destroy(list_patterns_fnmatch);
-      }
-      void prepare(c_strlist_t *excludes);
-
-      QRegularExpression regexp_exclude;
-      c_strlist_t *list_patterns_fnmatch = nullptr;
-
-  } parsed_traversal_excludes;
+  /**
+   * Function used to determine whether an item is excluded
+   * during the update phase.
+   *
+   * See ExcludedFiles in csync_exclude.
+   */
+  std::function<CSYNC_EXCLUDE_TYPE(const char *path, int filetype)> exclude_traversal_fn;
```

Comment:
I prefer tying ExcludedFiles and csync together through this function because it's more composable/decoupled. Now csync only ever cares about the traversal exclude match function which makes it easy to hook in a different exclude function for testing.